bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/Telegram.java

```diff
@@ -130,4 +143,107 @@ static public boolean sendTelegram(@ParamDoc(name = "group") String group,
 
 		return sendTelegram(group, String.format(format, args));
 	}
+
+	@ActionDoc(text = "Sends a Picture via Telegram REST API")
+	static public boolean sendTelegramPhoto(
+			@ParamDoc(name = "group") String group,
+			@ParamDoc(name = "photoURL") String photoURL,
+			@ParamDoc(name = "caption") String caption) {
+
+		if (groupTokens.get(group) == null) {
+			logger.error("Bot '{}' not defined, action skipped", group);
+			return false;
+		}
+
+		if (photoURL == null) {
+			logger.error("photoURL not defined, action skipped");
+			return false;
+		}
+
+		// load image from url
+		byte[] imageFromURL;
+
+		HttpClient getClient = new HttpClient();
+		GetMethod getMethod = new GetMethod(photoURL);
+		getMethod.getParams().setParameter(HttpMethodParams.RETRY_HANDLER,
+				new DefaultHttpMethodRetryHandler(3, false));
+		try {
+			int statusCode = getClient.executeMethod(getMethod);
+			if (statusCode != HttpStatus.SC_OK) {
+				logger.error("Method failed: {}", getMethod.getStatusLine());
+				return false;
+			}
+
+			imageFromURL = getMethod.getResponseBody();
+		} catch (HttpException e) {
+			logger.error("Fatal protocol violation: {}", e.toString());
+			return false;
+		} catch (IOException e) {
+			logger.error("Fatal transport error: {}", e.toString());
+			return false;
+		} finally {
+			getMethod.releaseConnection();
+		}
+
+		// parse image type
+		String imageType;
+		try {
+			ImageInputStream iis = ImageIO
+					.createImageInputStream(new ByteArrayInputStream(
+							imageFromURL));
+			Iterator<ImageReader> imageReaders = ImageIO.getImageReaders(iis);
+			if (!imageReaders.hasNext()) {
+				logger.error("photoURL does not represent a known image type");
+				return false;
+			}
+			ImageReader reader = (ImageReader) imageReaders.next();
+			imageType = reader.getFormatName();
+		} catch (IOException e) {
+			logger.error("cannot parse photoURL as image: {}", e.getMessage());
+			return false;
+		}
+
+		// post photo to telegram
+		String url = String.format(TELEGRAM_PHOTO_URL, groupTokens.get(group)
+				.getToken());
+
+		PostMethod postMethod = new PostMethod(url);
+		try {
+			postMethod.getParams().setContentCharset("UTF-8");
+			postMethod.getParams().setSoTimeout(HTTP_TIMEOUT);
+			postMethod.getParams().setParameter(HttpMethodParams.RETRY_HANDLER,
+					new DefaultHttpMethodRetryHandler(3, false));
+			Part[] parts = new Part[caption != null ? 3 : 2];
+			parts[0] = new StringPart("chat_id", groupTokens.get(group)
+					.getChatId());
+			parts[1] = new FilePart("photo", new ByteArrayPartSource(
+					String.format("image.%s", imageType), imageFromURL));
```

Comment:
Yes, your correct use of the file name extension to match the image type will make the generic "image" name unimportant to users.  This is a different scenario from attaching any arbitrary file to an email, where the name can actually matter.  Thanks for the detailed response.